### ISSUE TITLE
Iterations on W icon in the editor header

### ISSUE THREAD

**Issue by mtias** [WordPress/gutenberg#20929]
#20579 introduced a WordPress icon button to function as the "view posts / go back" action in full-screen mode.

- [x] Make it easy to extend / replace: this button should be customizable both in appearance and behaviour. @vindl https://github.com/WordPress/gutenberg/pull/22001 https://github.com/WordPress/gutenberg/pull/22179 https://github.com/WordPress/gutenberg/pull/22323
- [x] Render the current site icon (if one is set) instead of the WordPress one. @jeyip https://github.com/WordPress/gutenberg/pull/22952

![image](https://user-images.githubusercontent.com/548849/76769084-40539500-679c-11ea-9aa2-8829d2e80b82.png)

**Comment by nickhamze** [comment]
This my favorite GitHub issue of all time.  Thank you so much. The double ring logo makes me so sad. I can't wait to create all sorts of silly illustrations to replace that button with.

**Comment by tijosh** [comment]
+1 to this. It would be good to have the option to change it to anything, or at least have the possibility to change it back to a left/back arrow.

**Comment by folletto** [comment]
I agree on this one. I opened #21795 with a similar suggestion. More specifically, I think we should allow to:

1. Lock Gutenberg to be full-screen only
2. Change the page where the (W) goes back to 
3. Bonus: even better if the look can be customized too as noted above.

Also in regard to #21121, it's important that in case of modification of the "back" behaviour, the sidebar behaviour can be overridden.

The reason is that this would address some use cases that would allow Gutenberg to be way more integrated in various flows.

Use cases:
* Businesses could build custom workflows that open Gutenberg as their edit interface.
* Agencies could have their own interfaces to simplify the interaction with their clients.
* Products could use Gutenberg and make sure the user always come back to their UI.

In short, it should allow flows like:

Custom UI → Gutenberg → back to Custom UI

Bypassing WP Admin entirely, in order to avoid the user getting lost.

In short, it would make Gutenberg to be already more easily integrated.

**Comment by vindl** [comment]
Finished the first part of this issue with #22323.

**Comment by folletto** [comment]
Heya @vindl does that mean it's possible now to add a hook to override the destination of the W like I mentioned above, or not yet? 

I'm asking as I saw some work going on #22191 and I was wondering if it was already fixed, or if I should raise it again there.

**Comment by vindl** [comment]
> Heya @vindl does that mean it's possible now to add a hook to override the destination of the W like I mentioned above, or not yet?

Hi @folletto - This allows replacing the whole default component and the custom component can provide a different destination.

If we are looking into modifying just the destination while preserving everything else, that might require the rest of the functionality to be essentially copied over. I think that this can be improved in a similar fashion as I did for icon replacement case.

**Comment by folletto** [comment]
Ok so it's doable even if it requires right now to rewrite the component.

And yes, would be a nice improvement if there was a way to just switch the "back" action, without changing everything else :)

**Comment by vindl** [comment]
> And yes, would be a nice improvement if there was a way to just switch the "back" action, without changing everything else :)

I can look into that after #22191 gets merged in order to avoid introducing even more merge conflicts for them. :)

**Comment by folletto** [comment]
Thanks! :D

**Comment by vindl** [comment]
The second item has been resolved with merging of #22952.

**Comment by vindl** [comment]
> And yes, would be a nice improvement if there was a way to just switch the "back" action, without changing everything else :)

@folletto It took a while to circle back to this and we made some changes to how back button is handled in the site editor in the meantime (it's inside of nav menu, so a bit different than post editor). But this is now possible for both cases as of https://github.com/WordPress/gutenberg/pull/27213.

**Comment by folletto** [comment]
Thank you!
